healthworkforce/features/utils.js

```diff
@@ -0,0 +1,283 @@
+              "text": {
+                "status": "generated",
+                "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">USS Enterprise</div>"
+              },
+              "identifier": [
+                {
+                  "system": "http://www.acme.org.au/units",
+                  "value": "testLocation"
+                }
+              ],
+              "status": "active",
+              "name": locationName,
+              "mode": "instance"
+            }
+          }
+        ]
+      })
+}
+
+exports.gofrMockServiceOrganization = async () => {
+  await getResource('Organization')
+
+  nock('http://mock-service:4000')
+    .get('/gofr-organization-mock/_history')
+    .reply(
+      200,
+      {
+        "resourceType": "Bundle",
+        "type": "searchset",
+        "entry": [
+          {
+            "resourceType": "Organization",
+            "id": "123456",
+            "text": {
+              "status": "generated",
+              "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\">\n      \n      <p>Clinical Laboratory @ Acme Hospital. ph: +1 555 234 1234, email: \n        <a href=\"mailto:contact@labs.acme.org\">contact@labs.acme.org</a>\n      </p>\n    \n    </div>"
+            },
+            "identifier": [
+              {
+                "system": "http://www.acme.org.au/units",
+                "value": "testOrganization"
+              }
+            ],
+            "name": organizationName
+          }
+        ]
+      })
+}
+
+const getResource = resource => {
+  return axios({
+    url: `${OPENHIM_PROTOCOL}://${OPENHIM_API_HOSTNAME}:${OPENHIM_TRANSACTION_API_PORT}/hapi-fhir-jpaserver/fhir/${resource}?identifier:value=test${resource}`,
+    method: 'GET',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Basic ${authHeader}`
+    }
+  })
+}
+
+const removeResource = resource => {
+  return axios({
+    url: `${OPENHIM_PROTOCOL}://${OPENHIM_API_HOSTNAME}:${OPENHIM_TRANSACTION_API_PORT}/hapi-fhir-jpaserver/fhir/${resource}?identifier:value=test${resource}`,
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Basic ${authHeader}`
+    }
+  })
+}
+
+const verifyResourceDoesNotExist = async resource => {
+  const response = await getResource(resource)
+
+  expect(response.status).to.eql(200)
+
+  if (response.data.total > 0) throw Error(
+    `Test aborted! ${resource} resource used in test already exists and will be removed from the FHIR server`
+    )
+}
+
+exports.verifyPractitionerExistsAndCleanup = async () => {
+  const resource = 'Practitioner'
+  const response = await getResource(resource)
+
+  expect(response.status).to.eql(200)
+  expect(response.data.name[0].given, practitionerName)
+
+  await removeResource(resource)
+  .then(() => {console.log(`Test ${resource} successfully removed from FHIR`)})
+  .catch(() => {console.error(`Clean up failed, test ${resource} not removed from FHIR`)})
```

Comment:
If you are using `await` then you don't need to use `.then` and `.catch`. `await` will pause until it completes so any statement written after this is the same as doing a `.then`. For `.catch` you should wrap the entire block in a `try {} catch ()` statement.